notifications/notifications_test.go

```diff
@@ -0,0 +1,47 @@
+package notifications
+
+import (
+	"bytes"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/docker/distribution/context"
+	dNotifications "github.com/docker/distribution/notifications"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNotify(t *testing.T) {
+	err := Notify("NewThing", "GUN", dNotifications.SourceRecord{}, "username", &http.Request{}, dNotifications.NewBroadcaster())
+	require.NoError(t, err)
+}
+
+func TestNewRequestRecord(t *testing.T) {
+	req := &http.Request{
+		RemoteAddr: "https://remote.com",
+		Host:       "host",
+		Method:     "PUT",
+		Header: http.Header{
+			"User-Agent": []string{"ua"},
+		},
+	}
+	rec := newRequestRecord(req)
+	require.Equal(t, context.RemoteAddr(req), rec.Addr)
+	require.Equal(t, req.Host, rec.Host)
+	require.Equal(t, req.Method, rec.Method)
+	require.Equal(t, rec.UserAgent, "ua")
+}
+
+func TestCreateEvent(t *testing.T) {
+	req, err := http.NewRequest("GET", "https://url.com", bytes.NewBufferString("hi"))
+	require.Nil(t, err)
+
+	evt := createEvent("NewThing", "GUN", dNotifications.SourceRecord{Addr: "localhost"}, "username", req)
+	require.NotEmpty(t, evt.ID)
+	require.NotEqual(t, evt.Timestamp, time.Time{})
```

Comment:
I would rather not make tests time dependent. As long as the value is actually populated, it should be correct. It's a tradeoff, verifying times is not really possible to do correctly because we can't really mock out `time.Now()`.